c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -457,9 +461,92 @@ impl<'a, 'tcx> HirTyVisitor<'a, 'tcx> {
             }
         }
     }
+
+    pub fn gen_generics_rws<'p>(
+        &mut self,
+        generics: &Generics<'tcx>,
+        predicates: &ty::GenericPredicates<'tcx>,
+        origin_params: impl Iterator<Item = &'p OriginParam>,
+    ) {
+        let mut last_lifetime_span: Option<Span> = None;
+        let mut first_generic_type_span: Option<Span> = None;
+        let mut first_generic_const_span: Option<Span> = None;
+
+        for (predicate, predicate_span) in predicates.predicates {
+            if matches!(
+                predicate.kind().skip_binder(),
+                PredicateKind::RegionOutlives(OutlivesPredicate(a, b)) if a != b
+            ) && (last_lifetime_span.is_none()
+                || last_lifetime_span.unwrap().hi() < predicate_span.hi())
+            {
+                last_lifetime_span = Some(*predicate_span)
+            }
+        }
```

Comment:
I think this might malfunction on code with an `OutlivesPredicate` in a separate where clause:
```Rust
fn foo<'a, 'b>(x: &'a i32, y: &'b i32)
where 'a: 'b {
}
```
It will set `last_lifetime_span` to the end of the where clause rather than the end of the generic parameter list.

If you look at the HIR `generics.predicates` instead, the [`WhereRegionPredicate`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_hir/hir/struct.WhereRegionPredicate.html) case includes an `in_where_clause` flag you can use to filter these out.